File: ProyectoProg3/UtilsArmyModel/src/main/java/pe/edu/pucp/utilsarmy/gestion_de_material/model/Tesis.java
package pe.edu.pucp.utilsarmy.gestion_de_material.model;

public class Tesis extends MaterialBibliografico{
    private String institucionPublicacion;
    private String especialidad;
    private String asesor;
    private String grado; 
    
    public Tesis(){
        super();
    }

    public Tesis(String institucionPublicacion, String especialidad, String asesor, String grado, int idMaterial, int numero_copias, String titulo, int anho_publicacion, int numero_paginas, EstadoMaterial estado, String clasificacion_tematica, boolean activo, String idioma, Editorial editorial) {
<<<<<<< HEAD
        super(idMaterial, titulo, anho_publicacion, numero_paginas, estado, clasificacion_tematica, activo, idioma, editorial);
=======
        super(idMaterial, numero_copias, titulo, anho_publicacion, numero_paginas, estado, clasificacion_tematica, activo, idioma, editorial);
>>>>>>> e7cbe2292fd3facda28e5c7f4ace312400b5509f
        this.institucionPublicacion = institucionPublicacion;
        this.especialidad = especialidad;
        this.asesor = asesor;
        this.grado = grado;
    }

    public String getInstitucionPublicacion() {
        return institucionPublicacion;
    }

    public void setInstitucionPublicacion(String institucionPublicacion) {
        this.institucionPublicacion = institucionPublicacion;
    }

    public String getEspecialidad() {
        return especialidad;
    }

    public void setEspecialidad(String especialidad) {
        this.especialidad = especialidad;
    }

    public String getAsesor() {
        return asesor;
    }

    public void setAsesor(String asesor) {
        this.asesor = asesor;
    }

    public String getGrado() {
        return grado;
    }

    public void setGrado(String grado) {
        this.grado = grado;
    }
    
    
    
}

File: FrontEnd (C#)/SoftProgModel/GestMaterial/Tesis.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgModel.GestMaterial
{
    public class Tesis : MaterialBibliografico
    {
        private string institucionPublicacion;
        private string especialidad;
        private string asesor;
        private string grado;

        public Tesis(): base()
        {
        }

        public Tesis(string institucionPublicacion, string especialidad, string asesor, string grado, int idMaterial, int numero_copias, string titulo, 
            int anho_publicacion, int numero_paginas, EstadoMaterial estado, string clasificacion_tematica, bool activo, string idioma, TipoMaterial tipo):
            base (idMaterial, titulo, anho_publicacion, numero_paginas, estado, clasificacion_tematica, activo, idioma, tipo)
        {
            this.InstitucionPublicacion = institucionPublicacion;
            this.Especialidad = especialidad;
            this.Asesor = asesor;
            this.Grado = grado;
        }

        public string InstitucionPublicacion { get => institucionPublicacion; set => institucionPublicacion = value; }
        public string Especialidad { get => especialidad; set => especialidad = value; }
        public string Asesor { get => asesor; set => asesor = value; }
        public string Grado { get => grado; set => grado = value; }
    }
}
